opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectAttackPatternsKillChain.tsx

```diff
@@ -126,14 +133,14 @@ const StixDomainObjectAttackPatternsKillChain: FunctionComponent<StixDomainObjec
       .map((n) => n?.node);
   }
 
-  const killChains = data.attackPatterns ? Array.from(new Set([
-    'mitre-attack',
-    ...(data.attackPatterns.edges
-      .flatMap((edge) => edge.node.killChainPhases)
-      .map((phase) => phase?.kill_chain_name) ?? []),
-  ]))
-    .filter((name) => typeof name === 'string')
-    : [];
+  const killChainsData = usePreloadedFragment<AttackPatternsMatrixColumnsQuery, AttackPatternsMatrixColumns_data$key>({
+    queryDef: attackPatternsMatrixColumnsQuery,
+    fragmentDef: attackPatternsMatrixColumnsFragment,
+    queryRef: killChainDataQueryRef,
+  });
+  const killChainsPhaseData = killChainsData.attackPatternsMatrix?.attackPatternsOfPhases ?? [];
+  const killChains = R.uniq(killChainsPhaseData.map((a) => a.kill_chain_name))
+    .sort((a, b) => a.localeCompare(b));
```

Comment:
This works here's just a suggestion without Ramda (you can resolve if it's not relevant)
```suggestion
  const killChains = [...new Set(killChainsPhaseData.map((a) => a.kill_chain_name))]
    .sort((a, b) => a.localeCompare(b));
```

Reply:
we usually accept ramda for R.uniq, it is clearer with R.uniq I think